Storage/SQL/BrowserDB.swift

```diff
@@ -64,229 +51,158 @@ open class BrowserDB {
+    fileprivate func updateSchema(_ conn: SQLiteDBConnection, schema: Schema) -> Bool {
+        log.debug("Trying to update table '\(schema.name)' from version \(conn.version) to \(schema.version)")
+        if !schema.update(conn, from: conn.version) {
+            // If schema couldn't be updated, we'll bail without writing the schema version.
             log.debug("Updating failed.")
-            return .failed
+            return false
         }
 
-        var err: NSError? = nil
-
-        // Yes, we UPDATE OR INSERT… because we might be transferring ownership of a database table
-        // to a different Table. It'll trigger exists, and thus take the update path, but we won't
-        // necessarily have an existing schema entry -- i.e., we'll be updating from 0.
-        let insertResult = schemaTable.insert(conn, item: table, err: &err) ?? 0
-        let updateResult = schemaTable.update(conn, item: table, err: &err)
-        if  updateResult > 0 || insertResult > 0 {
-            return .updated
+        if let error = conn.setVersion(schema.version) {
+            log.error("Unable to update the schema version; \(error.localizedDescription)")
         }
-        return .failed
+
+        return true
     }
 
-    // Utility for table classes. They should call this when they're initialized to force
-    // creation of the table in the database.
-    func createOrUpdate(_ tables: Table...) -> DatabaseOpResult {
+    // Checks if the database schema needs created or updated and acts accordingly.
+    func prepareSchema(_ schema: Schema) -> DatabaseOpResult {
         guard !db.closed else {
-            log.info("Database is closed - skipping schema create/updates")
+            log.info("Database is closed; Skipping schema create or update.")
             return .closed
         }
 
-        var success = true
-
-        let doCreate = { (table: Table, connection: SQLiteDBConnection) -> Void in
-            switch self.createTable(connection, table: table) {
-            case .created:
-                success = true
-                return
-            case .exists:
-                log.debug("Table already exists.")
-                success = true
-                return
-            default:
-                success = false
-            }
+        // Get the current schema version for the database.
+        var currentVersion = 0
+        _ = self.db.withConnection(.readOnly, cb: { connection -> NSError? in
+            currentVersion = connection.version
+            return nil
+        })
+        
+        // If the current schema version for the database matches the specified
+        // `Schema` version, no further action is necessary and we can bail out.
+        if currentVersion == schema.version {
+            log.debug("Schema \(schema.name) already exists at version \(schema.version). Skipping additional schema preparation.")
+            return .success
         }
+        
```

Comment:
Er.. nevermind.. I'm gonna assume that we probably shouldn't because we'd effectively be put in a state where the app is stuck in an infinite startup crash loop.